### PR TITLE
Updated context to be used for HTTP request + refactoring

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

- Expanded the use of context to pass it to our HTTP requests.
- Added a context `nil` check in `azureblobstorage.go`.
- Re-named `cancelCtx` to `closeCtx` to better reflect the purpose of some of our `defer` calls.
- 1 loop refactoring to use the `for element := range someChannel` loops.

###### Does this affect the toolchain?  

No.

###### Associated issues  

- #9251

###### Test Methodology

- Local tests.
- [Full dev build](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=581443&view=results).

### CODE REVIEW

**Review by dmcilvaney — COMMENTED** [review]
While it seems slightly pointless, everywhere I look says to always `defer cancelFunc()`. By its nature it should always be safe to do so. In fact I believe `go vet` will actually flag any un-deferred cancel functions.

I think the best argument is that maybe today its ok, but what about  in 6 months? A developer will see a context, and assume that we followed best practices. They will add a go routine that only exits on the cancel signal, and now we have a memory leak.

tldr, `defer cancelFunc()` is a really easy and safe way to ensure that a context is cleaned up no mater what code flow we end up following.

File: toolkit/tools/internal/network/network_test.go [hunk 1]
```diff
@@ -139,9 +139,7 @@ func TestDownloadFile(t *testing.T) {
 			dstDir := t.TempDir()
```
Comment (dmcilvaney) [inline toolkit/tools/internal/network/network_test.go:139]: https://github.com/microsoft/azurelinux/pull/9307#discussion_r1626770961 We should add a test here to check early cancel propagates correctly.

**Review by dmcilvaney — APPROVED** [review]

**Review by sameluch — APPROVED** [review]
